plugins/modules/dcnm_image_upgrade.py

```diff
@@ -0,0 +1,5313 @@
+        Set the package_uninstall flag to True or False.
+
+        Default: False
+        """
+        return self.properties.get("package_uninstall")
+
+    @package_uninstall.setter
+    def package_uninstall(self, value):
+        name = "package_uninstall"
+        if not isinstance(value, bool):
+            msg = f"{self.class_name}.{name}.setter: "
+            msg += f"instance.{name} must be a boolean."
+            self.module.fail_json(msg)
+        self.properties[name] = value
+
+    @property
+    def reboot(self):
+        """
+        Set the reboot flag to True or False.
+
+        Default: False
+        """
+        return self.properties.get("reboot")
+
+    @reboot.setter
+    def reboot(self, value):
+        name = "reboot"
+        if not isinstance(value, bool):
+            msg = f"{self.class_name}.{name}.setter: "
+            msg += f"instance.{name} must be a boolean."
+            self.module.fail_json(msg)
+        self.properties[name] = value
+
+    @property
+    def write_erase(self):
+        """
+        Set the write_erase flag to True or False.
+
+        Default: False
+        """
+        return self.properties.get("write_erase")
+
+    @write_erase.setter
+    def write_erase(self, value):
+        name = "write_erase"
+        if not isinstance(value, bool):
+            msg = f"{self.class_name}.{name}.setter: "
+            msg += f"instance.{name} must be a boolean."
+            self.module.fail_json(msg)
+        self.properties[name] = value
+
+
+    # getter properties
+    @property
+    def check_interval(self):
+        """
+        Return the image upgrade check interval in seconds
+        """
+        return self.properties.get("check_interval")
+
+    @property
+    def check_timeout(self):
+        """
+        Return the image upgrade check timeout in seconds
+        """
+        return self.properties.get("check_timeout")
+
+    @property
+    def ndfc_data(self):
+        """
+        Return the data retrieved from NDFC for the image upgrade request.
+
+        instance.devices must be set first.
+        instance.commit() must be called first.
+        """
+        return self.properties.get("ndfc_data")
+
+    @property
+    def ndfc_result(self):
+        """
+        Return the POST result from NDFC
+        instance.devices must be set first.
+        instance.commit() must be called first.
+        """
+        return self.properties.get("ndfc_result")
+
+    @property
+    def ndfc_response(self):
+        """
+        Return the POST response from NDFC
+        instance.devices must be set first.
+        instance.commit() must be called first.
+        """
+        return self.properties.get("ndfc_response")
+
+    @property
+    def serial_numbers(self):
+        """
+        Return a list of serial numbers from self.devices
+        """
+        return [device.get("serial_number") for device in self.devices]
+
+
+class NdfcVersion(NdfcAnsibleImageUpgradeCommon):
```

Comment:
Is there a reason for having a separate class for getting NDFC version. We already have a function to retrieve this in module_utils. Did you check that?

Reply:
Yes, I did check it.  I didn't use it because there is no capability in that function to return patch version.  I needed patch version since image mgmt API changed between two patch versions.